server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/R4237DefaultRuleApplyingMigrate.java

```diff
@@ -0,0 +1,155 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.migrate.jdbc.common;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.oceanbase.odc.common.util.YamlUtils;
+import com.oceanbase.odc.core.migrate.JdbcMigratable;
+import com.oceanbase.odc.core.migrate.Migratable;
+import com.oceanbase.odc.core.shared.exception.UnexpectedException;
+import com.oceanbase.odc.metadb.regulation.ruleset.DefaultRuleApplyingEntity;
+import com.oceanbase.odc.metadb.regulation.ruleset.DefaultRuleApplyingRepository;
+import com.oceanbase.odc.metadb.regulation.ruleset.MetadataEntity;
+import com.oceanbase.odc.metadb.regulation.ruleset.RuleMetadataRepository;
+import com.oceanbase.odc.service.common.util.SpringContextUtil;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @Author: Lebie
+ * @Date: 2023/12/5 11:06
+ * @Description: []
+ */
+@Slf4j
+@Migratable(version = "4.2.3.7", description = "init default regulation rule applying", repeatable = true,
+        ignoreChecksum = true)
+public class R4237DefaultRuleApplyingMigrate implements JdbcMigratable {
+    private static final String MIGRATE_CONFIG_FILE = "init-config/init/regulation-rule-applying.yaml";
+
+    private DefaultRuleApplyingRepository defaultRuleApplyingRepository;
+    private RuleMetadataRepository ruleMetadataRepository;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void migrate(DataSource dataSource) {
+        this.defaultRuleApplyingRepository = SpringContextUtil.getBean(DefaultRuleApplyingRepository.class);
+        this.ruleMetadataRepository = SpringContextUtil.getBean(RuleMetadataRepository.class);
+        Map<String, MetadataEntity> metadataName2Metadata =
+                ruleMetadataRepository.findAll().stream().collect(Collectors.toMap(MetadataEntity::getName, e -> e));
+        List<InnerDefaultRuleApplying> expected =
+                YamlUtils.fromYaml(MIGRATE_CONFIG_FILE, new TypeReference<List<InnerDefaultRuleApplying>>() {});
+        Map<String, List<DefaultRuleApplyingEntity>> actualRulesetName2RuleApplyings = defaultRuleApplyingRepository
+                .findAll().stream().collect(Collectors.groupingBy(DefaultRuleApplyingEntity::getRulesetName));
+        List<DefaultRuleApplyingEntity> toAdd = new ArrayList<>();
+        for (InnerDefaultRuleApplying expectedApplying : expected) {
```

Comment:
expected will be null if the YamlUtils encounter an exception

Reply:
Okay, verification added.